spring/spring-data/src/test/java/io/quarkus/ts/spring/data/rest/PagingAndSortingRepositoryRestResourceIT.java

```diff
@@ -1,116 +1,61 @@
 import org.junit.jupiter.api.Test;
 
+import com.google.common.collect.Streams;
+
 import io.quarkus.test.scenarios.QuarkusScenario;
-import io.quarkus.ts.spring.data.AbstractDbIT;
-import io.restassured.response.Response;
 
 @QuarkusScenario
-class PagingAndSortingRepositoryRestResourceIT extends AbstractDbIT {
+class PagingAndSortingRepositoryRestResourceIT extends AbstractPagingAndSortingRepositoryRestResourceIT {
 
-    @Test
-    void testAllRepositoryMethods() throws InterruptedException {
-        //GET - List all articles (should have all 4 articles the database has initially)
-        app.given()
-                .accept("application/json")
-                .when().get("/articles")
-                .then()
-                .statusCode(200)
-                .body(
-                        containsString("Aeneid"),
-                        containsString("Beach House"),
-                        containsString("Cadillac Desert"),
-                        containsString("Dagon and Other Macabre Tales"));
+    private static final List<String> MODIFIED_ITEMS = Streams.concat(ORIGINAL_ITEMS.stream(), Stream.of(UPDATED_ITEM))
+            .collect(Collectors.toList());
 
-        //POST - Create a new Article
-        app.given()
-                .contentType("application/json")
-                .accept("application/json")
-                .body("{\"name\": \"Early Asimov\", \"author\": \"Isaac Asimov\"}")
-                .when().post("/articles")
-                .then()
-                .statusCode(201)
-                .body(containsString("Early Asimov"))
-                .body("id", notNullValue())
-                .extract().body().jsonPath().getString("id");
+    @Override
+    protected String getUrl() {
+        return "/articles";
+    }
 
-        //PUT - Update a new Article
-        app.given()
-                .contentType("application/json")
-                .accept("application/json")
-                .body("{\"name\": \"Early Asimov 2nd Edition\", \"author\": \"Isaac Asimov\"}")
-                .when().put("/articles/5")
-                .then()
-                .statusCode(204);
+    @Override
+    protected List<String> getItemsAfterUpdates() {
+        return MODIFIED_ITEMS;
+    }
 
-        //GET{id} - Find new article by id
-        app.given()
-                .when().get("/articles/id/5")
-                .then()
-                .statusCode(200)
-                .body(
-                        containsString("Early Asimov 2nd Edition"));
+    @Override
+    protected String getItemNameHalJsonPath() {
+        return "_embedded.articles.name";
+    }
 
-        //DELETE - Try to delete a article via HTTP (method not allowed)
-        app.given()
-                .when().delete("/articles/5")
-                .then()
-                .statusCode(405);
+    @Override
+    protected String getItemIdUrl(long id) {
+        return getUrl() + "/id/" + id;
+    }
 
-        //Test repository pagination
+    @Order(7)
```

Comment:
Both 7 and 9 are used in `AbstractRepositoryRestResourceIT.java`. Wouldn't create a conflict?

Reply:
No, it wouldn't. In fact, the overriden methods must re-declare the order.

However, I replaced the 2 overrides with 2 abstract helper methods. The implementations of the `delete` tests are the same, only the response statuses is different.